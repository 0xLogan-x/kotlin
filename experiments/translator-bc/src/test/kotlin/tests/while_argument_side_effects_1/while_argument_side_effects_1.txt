while_argument_side_effects_1() == 10099
