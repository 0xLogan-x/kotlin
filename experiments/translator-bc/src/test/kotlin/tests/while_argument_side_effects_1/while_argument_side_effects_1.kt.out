@exceptions.str.1 = private unnamed_addr constant  [59 x i8] c"Exception in thread main kotlin.KotlinNullPointerException\00", align 1
declare void @llvm.memcpy.p0i8.p0i8.i64(i8* nocapture, i8* nocapture readonly, i64, i32, i1)
declare i8* @malloc_heap(i32)
declare i32 @printf(i8*, ...)
declare void @abort()
%class.Nothing = type { }
%class.while_argument_side_effects_1_Counter = type { i32 }
@object.instance.while_argument_side_effects_1_Counter = internal global %class.while_argument_side_effects_1_Counter { i32 0 }, align 8
define weak void @while_argument_side_effects_1_Counter(%class.while_argument_side_effects_1_Counter*  %classvariable.this) #0
{
%classvariable.this.addr = alloca %class.while_argument_side_effects_1_Counter, align 8
%var2 = getelementptr inbounds %class.while_argument_side_effects_1_Counter* %classvariable.this.addr, i32 0, i32 0
%var3 = load i32* %var2, align 4
store i32 0, i32* %var2, align 4
%var4 = bitcast %class.while_argument_side_effects_1_Counter* %classvariable.this to i8*
%var5 = bitcast %class.while_argument_side_effects_1_Counter* %classvariable.this.addr to i8*
call void @llvm.memcpy.p0i8.p0i8.i64(i8* %var4, i8* %var5, i64 4, i32 4, i1 false)
ret void 
unreachable
}
define weak i1 @while_argument_side_effects_1_Counter.isEnd(%class.while_argument_side_effects_1_Counter*  %classvariable.this) #0
{
%var6 = getelementptr inbounds %class.while_argument_side_effects_1_Counter* %classvariable.this, i32 0, i32 0
%var7 = load i32* %var6, align 4
%var8 = load i32* %var6, align 4
%var9 = add nsw i32 %var8, 1
store i32 %var9, i32* %var6, align 4
%var10 = load i32* %var6, align 4
%var11 = getelementptr inbounds %class.while_argument_side_effects_1_Counter* %classvariable.this, i32 0, i32 0
%var12 = load i32* %var11, align 4
%var13 = icmp slt i32 %var12, 10
ret i1 %var13
unreachable
}
define weak i32 @while_argument_side_effects_1() #0
{
%managed.unique.0.while_argument_side_effects_1.iter = alloca i32, align 4
%var14 = load i32* %managed.unique.0.while_argument_side_effects_1.iter, align 4
store i32 0, i32* %managed.unique.0.while_argument_side_effects_1.iter, align 4
br label %label.while.1
label.while.1:
%var15 = call i1 @while_argument_side_effects_1_Counter.isEnd(%class.while_argument_side_effects_1_Counter* @object.instance.while_argument_side_effects_1_Counter)
%var16 = alloca i1, align 4
store i1 %var15, i1* %var16, align 4
%var17 = load i1* %var16, align 4
br i1 %var17, label %label.while.2, label %label.while.3
label.while.2:
%var18 = load i32* %managed.unique.0.while_argument_side_effects_1.iter, align 4
%var19 = load i32* %managed.unique.0.while_argument_side_effects_1.iter, align 4
%var20 = add nsw i32 %var19, 1
store i32 %var20, i32* %managed.unique.0.while_argument_side_effects_1.iter, align 4
%var21 = load i32* %managed.unique.0.while_argument_side_effects_1.iter, align 4
br label %label.while.1
label.while.3:
%var22 = getelementptr inbounds %class.while_argument_side_effects_1_Counter* @object.instance.while_argument_side_effects_1_Counter, i32 0, i32 0
%var23 = load i32* %var22, align 4
%var24 = mul nsw i32 1009, %var23
%var25 = load i32* %managed.unique.0.while_argument_side_effects_1.iter, align 4
%var26 = add nsw i32 %var25, %var24
ret i32 %var26
unreachable
}

